Gem/Code/Source/Weapons/WeaponGathers.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Source/Weapons/WeaponGathers.h>
+#include <Multiplayer/NetworkTime/INetworkTime.h>
+
+namespace MultiplayerSample
+{
+    AZ_CVAR(uint32_t, bg_MultitraceNumTraceSegments, 3, nullptr, AZ::ConsoleFunctorFlags::Null, "The number of segments to use when performing multitrace casts");
+
+    IntersectFilter::IntersectFilter
+    (
+        const AZ::Transform& initialPose, 
+        const AZ::Vector3& sweep, 
+        HitStatic intersectStatic, 
+        HitDynamic intersectDynamic, 
+        HitMultiple intersectMultiple, 
+        const NetEntityIdSet& filteredNetEntityIds
+    )
+        : m_initialPose(initialPose)
+        , m_sweep(sweep)
+        , m_intersectStatic(intersectStatic)
+        , m_intersectDynamic(intersectDynamic)
+        , m_intersectMultiple(intersectMultiple)
+        , m_filteredNetEntityIds(filteredNetEntityIds)
+    {
+        Multiplayer::INetworkTime* networkTime = Multiplayer::GetNetworkTime();
+        if (networkTime->IsTimeRewound())
+        {
+            m_rewindFrameId = networkTime->GetHostFrameId();
+        }
+    }
+
+    bool GatherEntities
+    (
+        const GatherParams& gatherParams, 
+        const ActivateEvent& eventData, 
+        const NetEntityIdSet& filteredNetEntityIds, 
+        [[maybe_unused]] IntersectResults& outResults
+    )
+    {
+        const AZ::Transform& startTransform = eventData.m_initialTransform;
+        const AZ::Vector3    sweep = eventData.m_targetPosition - startTransform.GetTranslation();
+        const HitMultiple    hitMultiple = gatherParams.m_multiHit ? HitMultiple::Yes : HitMultiple::No;
+        [[maybe_unused]] const GatherShape&   intersectShape = gatherParams.m_gatherShape;
+
+        IntersectFilter filter(startTransform, sweep, HitStatic::Yes, HitDynamic::Yes, hitMultiple, filteredNetEntityIds);
+        //gNovaGame->GetNetworkPhysicalWorld().WorldIntersect(intersectShape, filter, outResultList);
```

Comment:
This looks like it can be removed

Reply:
this one is critical, otherwise weapons won't be able to hit anything.. I just need this hooked up to the new scene query API for the physics gem, which I haven't had a chance to dig into yet